File: quicksort.py
import random
import timeit

def quicksort(tab):
    # First: scan table and put every item on the left of pivot if smaller else put it on the right
    # make subTable for left part of table and rerun quicksort on this
    # do the same for right part
    if len(tab) < 2:
        return tab
    leftSide = []
    rightSide = []
    equal = []
    pivot = tab[int(len(tab)/2)]
    for item in tab:
        if item != pivot:
            if item < pivot:
                leftSide.append(item)
            else:
                rightSide.append(item)
        else: 
            equal.append(item)
    return quicksort(leftSide) + equal + quicksort(rightSide)

def main():
    len = 10
    table = []
    for i in range(len):
        table.append(random.randint(0, 1000000))
    print("Not sorted : ", str(table))
    exectim = timeit.timeit(lambda: print("sorted : " + str(quicksort(table))), number=1)
    print("execution time = " + str(exectim))

if __name__ == "__main__":
    main()

File: quicksort.js
function getRandomInt(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min)) + min;
}

function quicksort(table) {
    if (table.length < 2) return table;
    let pivot = table[0];
    let rightSide = [];
    let leftSide = [];
    for (let i = 1; i < table.length; i++) {
        if (table[i] < pivot) {
            leftSide.push(table[i]);
        } else {
            rightSide.push(table[i]);
        }
    }
    return [...quicksort(leftSide), pivot, ...quicksort(rightSide)];
}

function main() {
    let table = [];
    let len = 10;
    for (let i = 0; i < len; i++) {
        table.push(getRandomInt(1, 100000));
    }
    console.log("Not sorted table : ", table);
    let sorted = quicksort(table);
    console.log("Sorted : ", sorted);
}

main();